benchmark/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+include(FetchContent)
+
+FetchContent_Declare(
+    benchmark
+    GIT_REPOSITORY https://github.com/google/benchmark.git
+    GIT_TAG        v1.5.3
+)
+FetchContent_GetProperties(benchmark)
+if(NOT benchmark_POPULATED)
+    set(BENCHMARK_ENABLE_GTEST_TESTS OFF)
+    set(BENCHMARK_ENABLE_INSTALL OFF)
+    set(BENCHMARK_ENABLE_TESTING OFF)
+    FetchContent_Populate(benchmark)
+    add_subdirectory(${benchmark_SOURCE_DIR} ${benchmark_BINARY_DIR})
```

Comment:
we could do:
```
    # disable warnings for both targets inside the google `benchmark` build
    rigel_disable_warnings(benchmark)
    rigel_disable_warnings(benchmark_main)
```
but I would steer clear of this one since it's quite brittle. 
do we care if third party code has warnings when it builds itself? I would say no.
the better option is to have granular compiler warnings and set it ONLY on RigelEngine code, not on third party stuff.

Reply:
Yeah that sounds good and cleaner/easier to maintain 👍🏻

Reply:
These changes are present in the latest commit.
I'll resolve this since I started it and the changes are there.

Reply:
Sure thing! 👍🏻 Thanks for making the changes!